kentik_api_library/setup.py

```diff
@@ -137,6 +128,50 @@ def _isort(self) -> None:
         run_cmd(cmd, self.announce)
 
 
+class FetchGRPCCode(Command):
+    """Command copying generate Python gRPC code from source repo."""
+
+    description = "Copy generated Python stubs from source repo"
+    user_options = [
+        ("repo=", None, "Source repository"),
+        ("src-path=", None, "Path to generated Python code within the source repo"),
+        ("dst-path=", None, "Destination path in the local source tree"),
+    ]
+
+    def initialize_options(self):
+        self.repo = "https://github.com/kentik/api-schema-public.git"
+        self.src_path = "gen/python"
+        self.dst_path = HERE.joinpath("kentik_api").joinpath("generated").as_posix()
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        print("Fetching gRPC generated code")
+
+        import git
+
+        # cleanup destination directory
+        shutil.rmtree(self.dst_path, ignore_errors=True)  # ignore "No such file or directory"
+        # create destination directory, if it does not exist
+        dst = Path(self.dst_path)
+        dst.mkdir(parents=True)
+        # checkout source repo and copy stubs
+        with TemporaryDirectory() as tmp:
+            git.Repo.clone_from(self.repo, tmp)
+            Path(tmp).joinpath(self.src_path).rename(dst)
+
+
+def list_packages(root: str) -> List[str]:
+    print("Listing packages")
```

Comment:
Use `setuptools.find_packages`. Your version is slightly broken, because it does not test that there is `__init__.py` in the directory before declaring it to be a package.

I actually slightly prefer to declare packages statically in `setup.py`. It leads to fewer surprises (e.g. including unit test packages in the list) ;). ... and there is really no need to install unit tests if somebody wants to  just use the SDK (as opposed to contribute to development), so they should not be in the `packages` list.

BTW, why do you want to print the packages while constructing the list?

Reply:
Changing to static package list